### DISCUSSION TITLE
How to add saml extension element to IDP AuthnRequest?

### DISCUSSION THREAD

**Discussion by pohjamat** [keycloak/keycloak#15882] [Q&A]
Hello,
I’m integrating keycloak 20 by saml to IDP that accepts saml2:Extensions element for authentication page language, for example:
…

  <samlp:Extensions>
        <vetuma xmlns="urn:vetuma:SAML:2.0:extensions">
            <LG>en</LG>
        </vetuma>
    </samlp:Extensions>
</samlp:AuthnRequest>
How could I configure keycloak identity provider to add this extension element to AuthRequest?

In keycloak admin console, realm -> Identity providers -> Provider details -> Settings tab, I can see 'Requested AuthnContext Constraints' section, but I cannot see how to add extension element to request.
Has someone done similar?

regards, Matti

**Comment by pohjamat** [comment]
sorry, the extension element looks like this
```
<samlp:Extensions>
        <vetum xmlns="urn:vetum:SAML:2.0:extensions">
            <LG>en</LG>
        </vetum>
    </samlp:Extensions>
</samlp:AuthnRequest>
```

**Reply by Albertsen96** [reply]
@pohjamat Did you ever find a solutiuon for this?

**Comment by pohjamat** [comment]
No, did not find a solution.

Lähettäjä: Albertsen96 ***@***.***>
Päivämäärä: tiistaina, 1. lokakuuta 2024 klo 17.13
Vastaanottaja: keycloak/keycloak ***@***.***>
Kopio: Matti Pohjalainen ***@***.***>, Mention ***@***.***>
Aihe: Re: [keycloak/keycloak] How to add saml extension element to IDP AuthnRequest? (Discussion #15882)

@pohjamat<https://github.com/pohjamat> Did you ever find a solutiuon for this?

**Comment by JirkaNovak** [comment]
I implemented a similar solution for an eIDAS IdP using a custom implementation of the SamlAuthenticationPreprocessor. The extension is written in Java. In the beforeProcessingLoginRequest method, the required extensions are collected, and in the beforeSendingLoginRequest method, they are added to the IdP request.
